samples/dml.js

```diff
@@ -0,0 +1,666 @@
+      })
+      .then(() => {
+        // Close the database when finished.
+        return database.close();
+      });
+  });
+  // [END spanner_dml_standard_update_with_timestamp]
+}
+
+function writeAndReadUsingDml(instanceId, databaseId, projectId) {
+  // [START spanner_dml_write_then_read]
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  database.runTransaction((err, transaction) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    transaction
+      .runUpdate({
+        sql: `INSERT Singers (SingerId, FirstName, LastName)
+          VALUES (11, 'Timothy', 'Campbell')`,
+      })
+      // Queries rows from the Singers table
+      .run(`SELECT FirstName, LastName FROM Singers`)
+      .then(results => {
+        const rows = results[0];
+        rows.forEach(row => {
+          const json = row.toJSON();
+          console.log(`${json.FirstName} ${json.LastName}`);
+        });
+        return transaction.commit();
+      })
+      .catch(err => {
+        console.error('ERROR:', err);
+      })
+      .then(() => {
+        // Close the database when finished.
+        return database.close();
+      });
+  });
+  // [END spanner_dml_write_then_read]
+}
+
+function updateUsingDmlWithStruct(instanceId, databaseId, projectId) {
+  // [START spanner_dml_structs]
+  const nameStruct = Spanner.struct({
+    FirstName: 'Timothy',
+    LastName: 'Campbell',
+  });
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance and database
+  const instance = spanner.instance(instanceId);
+  const database = instance.database(databaseId);
+
+  database.runTransaction((err, transaction) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    transaction
+      .runUpdate({
+        sql: `UPDATE Singers SET LastName = 'Grant'
+        WHERE STRUCT<FirstName STRING, LastName STRING>(FirstName, LastName) = @name`,
+        params: {
+          name: nameStruct,
+        },
+      })
+      .then(rowCount => {
+        console.log(`Successfully updated ${rowCount} record.`);
```

Comment:
should this be `records.`? Since record is correct only for 1==rowCount?

Reply:
This is intended as a single record update.